shoppingcart/workflow.go

```diff
@@ -0,0 +1,45 @@
+package shoppingcart
+
+import (
+	"go.temporal.io/sdk/workflow"
+)
+
+var (
+	UpdateName    = "shopping-cart"
+	TaskQueueName = "shopping-cart-tq"
+)
+
+type CartSignalPayload struct {
+	Action   string `json:"action"` // "add" or "remove"
+	ItemID   string `json:"item_id"`
+	Quantity int    `json:"quantity"`
+}
+
+type CartState map[string]int // itemID -> quantity
```

Comment:
nit: I would make `CartState` a struct with a map inside. I suggested this because it would be more externsible in the future if we wanted to add additional data to the cart state.